boto/ec2/connection.py

```diff
@@ -4423,6 +4424,11 @@ def copy_image(self, source_region, source_image_id, name=None,
             params['Description'] = description
         if client_token is not None:
             params['ClientToken'] = client_token
+        if encrypted is not None:
+            params['Encrypted'] = 'true' if encrypted else 'false'
+
+            if kms_key_id is not None:
```

Comment:
I think we should add `kms_key_id` if the user provides it, regardless of whether or not they specify `encrypted`.  If the user mistakenly adds `kms_key_id` without `encrypted` then we silently ignore `kms_key_id` and the user may not know anything is wrong.  If we always add the `kms_key_id` they'll at least get an error from EC2 saying that you have to include `encrypted` if you're including `kms_key_id`.